docs/decisions/0003-application-recomposition.rst

```diff
@@ -0,0 +1,166 @@
+        ...
+        Head/
+        learner-record/
+            index.js
+            index.scss
+            ProgramRecord/
+            ProgramRecordAlert/
+            <other view components>
+
+        verifiable-credentials/
+            index.js
+            <view components>
+
+#. Being a single feature Learner Records was responsible for some generic UI pieces. Now we'll introduce a top level navigation which pushes those generic UI to be extracted from the Learner Records.
+
+    * a new top level page component will be introduced (generic navigation actions carrying):
+        * "Back to My Profile" navigation
+        * features tabs switching (new UI/UX)
+    * ProgramRecordList component responsibilities will be reduced to its own functions:
+        * Record items rendering (with helpers);
+        * main container rendering goes away;
+        * back-to-profile navigation goes away;
+
+#. Data management will be extracted under ``data`` directories.
+
+    .. note::
+        We can postpone Learner Record data management refactoring for now (as a tech debt).
+
+    ::
+
+        assets/
+        i18n/
+        ...
+        Head/
+        learner-record/
+            data/
+            index.js
+            index.scss
+            ProgramRecord/
+            ProgramRecordAlert/
+            <other view components>
+
+        verifiable-credentials/
+            data/
+            index.js
+            index.scss
+            <view components>
+
+Routing
+-------
+
+Current application routes map is designed as a Learner Records feature-centric:
+
+    ::
+
+        / => ProgramRecordsList
+        /:programUUID => ProgramRecord (private)
+        /shared/:programUUID => ProgramRecord (public)
+
+Having other features requires additional namespacing:
+
+    ::
+
+        / => /program-records/
+        /program-records/ => ProgramRecordsList
+        /program-records/:programUUID => ProgramRecord (private)
+        /program-records/shared/:programUUID => ProgramRecord (public)
+        /verifiable-credentials/
+
+    * Learner Records routes (all current routes) go under ``/program-records`` path;
+    * Verifiable Credentials routes are added under ``/verifiable-credentials`` path;
+    * root (main home) route is set as redirect to the default feature (Learner Records home);
+
+State management
+----------------
+
+Currently, Redux store isn't used at all - everything is managed via local React state. That was Ok for the initial steps, but having another feature we should start global state management. `We can leverage a lot from the Redux (RTK)`_.
+
+Services (API)
+--------------
+
+.. note::
+    This item continues the State Management point.
+
+At this point, micro-frontend has a pretty few interactions (data fetches) with a single backend API. We use ``getAuthenticatedHttpClient`` on a low level (e.g. directly, each time configured). We than manually manage API request results (e.g. error checking, loading states, etc.).
```

Comment:
edit suggestion: At this point, the micro-frontend has a few interactions (data fetches) with a single backend API. We use ``getAuthenticatedHttpClient`` on a low level (e.g. directly,  each time it's configured), and then handle the API request results (e.g. error checking, loading states, etc.).